projects/layer-selector/src/lib/layer-selector.component.ts

```diff
@@ -0,0 +1,364 @@
+import { Component, ElementRef, Input, OnInit, ViewEncapsulation } from '@angular/core';
+import Basemap from 'arcgis-js-api/Basemap';
+import FeatureLayer from 'arcgis-js-api/layers/FeatureLayer';
+import LOD from 'arcgis-js-api/layers/support/LOD';
+import TileInfo from 'arcgis-js-api/layers/support/TileInfo';
+import WebTileLayer from 'arcgis-js-api/layers/WebTileLayer';
+
+import { setTileInfosForApplianceLayers } from './Discover';
+import { createDefaultTileInfo } from './TileInfo';
+
+const imageryAttributionJsonUrl = 'https://mapserv.utah.gov/cdn/attribution/imagery.json';
+
+@Component({
+  selector: 'agrc-layer-selector',
+  templateUrl: './layer-selector.component.html',
+  styleUrls: ['./layer-selector.component.scss'],
+  encapsulation: ViewEncapsulation.None
+})
+export class LayerSelectorComponent implements OnInit {
+  @Input()
+  makeExpandable: true;
+  @Input()
+  baseLayers = [];
+  @Input()
+  overlays = [];
+  @Input()
+  view: any;
```

Comment:
should we add the esri types to this since we can?

Reply:
i guess this is moot since the web pack config doesn't quite jive with the library building?

Reply:
Actually, I think that the type definitions is a separate thing from the custom webpack build. I bet that they could still be added.

Reply:
i think it's worth checking out to see how it works then